packages/typespec-client-generator-core/test/decorators/mark-as-pageable.test.ts

```diff
@@ -0,0 +1,228 @@
+import { ok, strictEqual } from "assert";
+import { beforeEach, it } from "vitest";
+import { SdkTestRunner, createSdkTestRunner } from "../test-host.js";
+
+let basicRunner: SdkTestRunner;
+
+beforeEach(async () => {
+  basicRunner = await createSdkTestRunner({
+    emitterName: "@azure-tools/typespec-csharp",
+  });
+});
+
+it("should mark regular operation as pageable when decorated with @markAsPageable", async () => {
+  await basicRunner.compile(`
+      @service
+      namespace TestService {
+        model ItemListResult {
+          @pageItems
+          items: Item[];
+          @nextLink
+          nextLink?: string;
+        }
+
+        model Item {
+          id: string;
+          name: string;
+        }
+
+        @Azure.ClientGenerator.Core.Legacy.markAsPageable
+        @route("/items")
+        @get
+        op listItems(): ItemListResult;
+      }
+    `);
+
+  const methods = basicRunner.context.sdkPackage.clients[0].methods;
+  strictEqual(methods.length, 1);
+
+  const method = methods[0];
+  strictEqual(method.kind, "paging");
+  strictEqual(method.name, "listItems");
+
```

Comment:
we should also assert the `pageableMetadata` also has the correct information here?